src/main/java/com/example/controllers/FilterController.java

```diff
@@ -0,0 +1,123 @@
+package com.example.controllers;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import com.example.repository.CurrencyRepository;
+
+import model.Currency;
+
+@RequestMapping("/filter")
+@RestController
+public class FilterController {
+
+	@Autowired
+	CurrencyRepository cr;
+
+	// Just for testing
+	public void printCurrency(Currency c) {
+		System.out.println(c.getName().trim() + "\t" + c.getSymbol().trim() + "\t" + c.getPriceusd() + "\t"
+				+ c.getChangepercent24h() + "\t" + c.getThreshold());
+	}
+
+	@RequestMapping(value = "/filterByRange", method = RequestMethod.GET)
+	public List<Currency> filterByRange(@RequestParam(value = "min") float minRange,
+			@RequestParam(value = "max") float maxRange) {
+
+		List<Currency> allCurrencies = cr.findAll();
+		List<Currency> filteredList = new ArrayList<Currency>();
+
+		for (Currency c : allCurrencies) {
+
+			if (c.getPriceusd() > minRange && c.getPriceusd() < maxRange) {
+				filteredList.add(c);
+				printCurrency(c);
+
+			}
+		}
+
+		return filteredList;
+
+	}
+
+	@RequestMapping(value = "/filterTop3", method = RequestMethod.GET)
+	public List<Currency> filterTop3(@RequestParam(value = "filter") String filter,
+			@RequestParam(value = "method") String m) {
```

Comment:
Variable and parameter names like 'm' should be avoided, to make the code more readable. It should be clear what it represents later in the code.